pyface/data_view/abstract_value_type.py

```diff
@@ -182,6 +182,50 @@ def set_text(self, model, row, column, text):
         """
         raise DataViewSetError("Cannot set value.")
 
+    def has_image(self, model, row, column):
```

Comment:
There is a distinction in most backends between an image and an icon - an icon is a collection of related images with different styles (ie. versions for enabled/disabled, highlighted, activated, etc.).  This is very much _not_ an icon in this sense, and Pyface has no object which corresponds to an icon (`IImageResource` tries to be both an image and an icon and fails at both).  So these method as written are very much a "image"-based.

Let's leave this as-is for now, since this is currently the only way to display image data; we might revise when it comes time to implement custom editors/renderers and provide a separate channel for icon-like image collections.

Also note that Qt decorations are not limited to icons, and can potentially display thumbnails or even full images (as well as colors).

Reply:
Okay. Just trying to say that as someone reading this method's docstring, it is not obvious that it is used as a decoration rather than an image filling up the whole cell. It was a bit of a surprise that the editable value is used alongside this image.

Qt documentation is very specific about what `DecorationRole` means:
> The data to be rendered as a decoration in the form of an icon. (QColor, QIcon or QPixmap)

Reply:
If you have a large image and no text, the decoration _will_ fill up the whole cell (or most of it - I think there may be a border, and you have to take into account how cell sizes are determined).

As someone writing an implementation of this method, you _shouldn't_ care how this image is being displayed; you are providing some image data, and it is up to the view to determine what to do with it.